src/send/EnterAmount.tsx

```diff
@@ -244,17 +279,22 @@ function EnterAmount({
     !showLowerAmountError &&
     prepareTransactionsResult &&
     prepareTransactionsResult.type === 'not-enough-balance-for-gas'
-  const sendIsPossible =
+  const transactionIsPossible =
     !showLowerAmountError &&
     prepareTransactionsResult &&
     prepareTransactionsResult.type === 'possible' &&
     prepareTransactionsResult.transactions.length > 0
 
+  const disabled =
+    disableProceed ||
+    (disableBalanceCheck && tokenAmount?.isZero()) ||
+    (!disableBalanceCheck && !transactionIsPossible)
```

Comment:
maybe this is slightly more readable? don't feel strongly though
```suggestion
    (disableBalanceCheck ? tokenAmount?.isZero() : !transactionIsPossible)
```